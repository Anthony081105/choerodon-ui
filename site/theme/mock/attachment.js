import Mock from 'mockjs';

const dataSetAttachmentTemple = [
  {
    fileId: 1,
    fileName: 'demo1.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-16 19:19:00',
  },
  {
    fileId: 2,
    fileName: 'demo2.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-17 19:19:00',
  },
];

const dataSetAttachmentTemple2 = [
  ...dataSetAttachmentTemple,
  {
    fileId: 3,
    fileName: 'demo3.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-18 19:19:00',
  },
  {
    fileId: 4,
    fileName: 'demo4.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
  {
    fileId: 5,
    fileName: 'demo5.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
  {
    fileId: 6,
    fileName: 'demo6.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
  {
    fileId: 7,
    fileName: 'demo7.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
  {
    fileId: 8,
    fileName: 'demo8.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
  {
    fileId: 9,
    fileName: 'demo9.png',
    fileType: 'image/png',
    fileSize: 300000,
    fileUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
    creationDate: '2021-08-19 19:19:00',
  },
];

const dataSetAttachmentCountTemple = {
  '4c74a34a-fa37-4e92-be9d-5cf726fb1472': 2,
};

const dataSetAttachmentCountTemple2 = {
  '88888888-fa37-4e92-be9d-5cf726fb1472': 9,
};

const dataSetAttachmentCountTemple3 = {
  '4c74a34a-fa37-4e92-be9d-5cf726fb1472': 2,
  '88888888-fa37-4e92-be9d-5cf726fb1472': 9,
};

const dataSetAttachmentData = Mock.mock(dataSetAttachmentTemple);
const dataSetAttachmentData2 = Mock.mock(dataSetAttachmentTemple2);
const dataSetAttachmentCount = Mock.mock(dataSetAttachmentCountTemple);
const dataSetAttachmentCount2 = Mock.mock(dataSetAttachmentCountTemple2);
const dataSetAttachmentCount3 = Mock.mock(dataSetAttachmentCountTemple3);

const dataSetAttachmentRule = /\/attachment\/4c74a34a-fa37-4e92-be9d-5cf726fb1472/;
const dataSetAttachmentRule2 = /\/attachment\/88888888-fa37-4e92-be9d-5cf726fb1472/;
const dataSetAttachmentCountRule = /\/attachment-count\/4c74a34a-fa37-4e92-be9d-5cf726fb1472$/;
const dataSetAttachmentCountRule2 = /\/attachment-count\/88888888-fa37-4e92-be9d-5cf726fb1472$/;
const dataSetAttachmentCountRule3 = /\/attachment-count\/4c74a34a-fa37-4e92-be9d-5cf726fb1472,88888888-fa37-4e92-be9d-5cf726fb1472$/;

export default function () {
  if (typeof window !== 'undefined') {
    Mock.setup({ timeout: 0 });

    Mock.mock(dataSetAttachmentRule, dataSetAttachmentTemple);

    Mock.mock(dataSetAttachmentRule2, dataSetAttachmentTemple2);

    Mock.mock(dataSetAttachmentCountRule, dataSetAttachmentCountTemple);

    Mock.mock(dataSetAttachmentCountRule2, dataSetAttachmentCountTemple2);

    Mock.mock(dataSetAttachmentCountRule3, dataSetAttachmentCountTemple3);
  }
}

export const attachmentTempleList = [
  { rule: dataSetAttachmentRule, data: dataSetAttachmentData },
  { rule: dataSetAttachmentRule2, data: dataSetAttachmentData2 },
  { rule: dataSetAttachmentCountRule, data: dataSetAttachmentCount },
  { rule: dataSetAttachmentCountRule2, data: dataSetAttachmentCount2 },
  { rule: dataSetAttachmentCountRule3, data: dataSetAttachmentCount3 },
];
