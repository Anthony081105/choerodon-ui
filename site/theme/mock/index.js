import dataset, { dsTempleList } from './dataset';
import tree, { treeTempList } from './tree';
import lov, { lovTempleList } from './lov';
import lookup, { lookupTempleList } from './lookup';
import upload, { uploadTempList } from './upload';
import attachment, { attachmentTempleList } from './attachment';

export default function () {
  dataset();
  tree();
  lov();
  lookup();
  upload();
  attachment();
  if (typeof window !== 'undefined') {
    const XHR = window._XMLHttpRequest;
    if (XHR) {
      const xhr = new XHR();
      window.XMLHttpRequest.prototype.upload = xhr.upload;
    }
  }
}

export const mockTempleList = [
  ...treeTempList,
  ...lovTempleList,
  ...lookupTempleList,
  ...dsTempleList,
  ...uploadTempList,
  ...attachmentTempleList,
];
