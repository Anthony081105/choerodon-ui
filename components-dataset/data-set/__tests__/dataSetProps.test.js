import React from 'react';
import { render, shallow, mount } from 'enzyme';

import DataSet from '..';
import fakeLargeData from '../../../site/theme/mock/performance-data/fakeLargeData.json';
import Table from '../../../components-pro/table';
import Record from '../Record';
import { randomData } from './mock'

const testName = 'testName';
const data = [
  {
    "userid": 1,
    "name": "假数据宋江",
    "sex": "M",
    "age": 68,
    "email": "6888888@qq.com",
    "active": true
  },
  {
    "userid": 2,
    "name": "假数据张飞",
    "sex": "M",
    "age": 60,
    "email": "zhangfei@163.com",
    "active": true
  },
  {
    "userid": 3,
    "name": "假数据赵云",
    "sex": "M",
    "age": 25,
    "email": "zhaoyun@163.com",
    "active": true
  },
  {
    "userid": 4,
    "name": "假数据廉颇",
    "sex": "M",
    "age": 88,
    "email": "61238@google.com",
    "active": false
  },
  {
    "userid": 5,
    "name": "假数据秦秀英",
    "sex": "F",
    "age": 12,
    "email": "6qxy8@hand-china.com",
    "active": true
  },
  {
    "userid": 6,
    "name": "假数据孔秀兰",
    "sex": "F",
    "age": 24,
    "email": "123368@qq.com",
    "active": true
  },
  {
    "userid": 7,
    "name": "假数据马芳",
    "sex": "F",
    "age": 10,
    "email": "mafangs@163.com",
    "active": false
  },
  {
    "userid": 8,
    "name": "假数据鲁智深",
    "sex": "M",
    "age": 48,
    "email": "6handlzs8@hand-china.com",
    "active": true
  }
];
const commonDs = {
  primaryKey: 'userid',
  fields: [
    {
      name: 'userid',
      type: 'number',
      label: '编号',
      required: true,
    },
    {
      name: 'name',
      type: 'string',
      label: '姓名',
    },
    {
      name: 'age',
      type: 'number',
      label: '年龄',
    },
    {
      name: 'sex',
      type: 'string',
      label: '性别',
      lookupCode: 'HR.EMPLOYEE_GENDER',
      required: true,
    },
    { name: 'email', type: 'string', label: '邮箱' },
    { name: 'active', type: 'boolean', label: '在线' },
  ],
  transport: {
    read: () => {
      return {
        // 此处设置的url无效，但是官网是可行的
        url: 'https://www.fastmock.site/mock/423302b318dd24f1712751d9bfc1cbbc/mock/guide/user',
        method: 'GET',
      };
    }
  },
};
const treeDs = {
  primaryKey: 'userid',
  autoQuery: true,
  parentField: 'parentId',
  expandField: 'expand',
  idField: 'userid',
  fields: [
    { name: 'userid', type: 'number' },
    { name: 'expand', type: 'boolean' },
    { name: 'parentId', type: 'number' },
  ],
  data: [
    { userid: 1, expand: true },
    { userid: 11, expand: true, parentId: 1 },
    { userid: 12, expand: true, parentId: 1 },
    { userid: 13, expand: true, parentId: 1 },
  ],
};
const commonQueryFields = [
  { name: 'enable', type: 'boolean', label: '是否开启' },
  { name: 'name', type: 'string', label: '姓名', defaultValue: 'Hugh' },
  { name: 'age', type: 'number', label: '年龄' },
]
const commonColumns = [
  { name: 'userid' },
  { name: 'name', editor: true },
  { name: 'age', editor: true },
  { name: 'sex', editor: true },
  { name: 'email', editor: true },
  { name: 'active', editor: true },
];

describe('DataSetProps', () => {
  describe('name', () => {
    it('name-自动生成 submitUrl，queryUrl，tlsUrl，validateUrl 正确性校验', () => {
      const ds = new DataSet({ ...commonDs, name: testName });
      expect(ds.tlsUrl).toBe(`/dataset/${testName}/languages`);
      expect(ds.validateUrl).toBe(`/dataset/${testName}/validate`);
      expect(ds.queryUrl).toBe(`/dataset/${testName}/queries`);
      expect(ds.submitUrl).toBe(`/dataset/${testName}/mutations`);
      expect(ds.exportUrl).toBe(`/dataset/${testName}/export`);
    })
  });

  describe('data', () => {
    it('data-验证初始化数据的数量及其数据字段的正确性', () => {
      const ds = new DataSet({ ...commonDs, data: data });
      if (ds.pageSize < data.length) {
        expect(ds.originalData.length).toBe(ds.pageSize);
      } else {
        expect(ds.originalData.length).toBe(data.length);
      }
      ds.originalData.map((record, i) => {
        expect(record.data).toEqual(data[i]);
      })
    })
  });

  describe('autoCreate', () => {
    it('autoCreate: true, autoQuery: false, 是否自动创建记录', () => {
      const ds = new DataSet({ ...commonDs, autoCreate: true, autoQuery: false })
      expect(ds.records.length).toBe(1);
      expect(ds.get(0)).toBeInstanceOf(Record);
    })
    it('autoCreate: false, autoQuery: false, 是否自动创建记录', () => {
      const ds = new DataSet({ ...commonDs, autoCreate: false, autoQuery: false })
      expect(ds.records.length).toBe(0);
      expect(ds.get(0)).toBeUndefined();
    })
    it('autoCreate: true, autoQuery: false, 存在初始数据, 是否自动创建记录', () => {
      const ds = new DataSet({ ...commonDs, autoCreate: true, autoQuery: false, data: data })
      expect(ds.records.length).toBe(data.length);
      ds.originalData.map((record, i) => {
        expect(record.data).toEqual(data[i]);
      })
      expect(ds.get(0)).not.toBeUndefined();
    })
    it('autoCreate: false, autoQuery: false, 存在初始数据, 是否自动创建记录', () => {
      const ds = new DataSet({ ...commonDs, autoCreate: false, autoQuery: false, data: data })
      expect(ds.records.length).toBe(data.length);
      ds.originalData.map((record, i) => {
        expect(record.data).toEqual(data[i]);
      })
      expect(ds.get(0)).not.toBeUndefined();
    })
  });

  describe('autoLocateFirst', () => {
    it('autoLocateFirst开启-多条初始数据', () => {
      const ds = new DataSet({ ...commonDs, autoLocateFirst: true, data: data });
      expect(ds.current).toBeInstanceOf(Record);
      expect(ds.get(0).isCurrent).toBeTruthy();
    })
    it('autoLocateFirst关闭-多条初始数据', () => {
      const ds = new DataSet({ ...commonDs, autoLocateFirst: false, data: data });
      expect(ds.current).toBeUndefined();
      expect(ds.get(0).isCurrent).toBeUndefined();
    })
  });

  // TODO: ①死数据无法翻页 ②由于弹出框的渲染位置并不在mount挂载的元素内部，因此find方法找不到
  // issue: https://github.com/enzymejs/enzyme/issues/2389
  // describe('modifiedCheck', () => {
  //   it('modifiedCheck: true,有记录更改,验证是否有警告提示', () => {
  //     const ds = new DataSet({ ...commonDs, modifiedCheck: true, data: randomData(30) });
  //     window.ds = ds;

  //     ds.get(0).set('name','changedName');
  //     ds.page(2);
  //     // const bodyEle = document.querySelector('body');
  //     const bodyEle = document.body.querySelector('div');
  //     const wrapper = mount(bodyEle);
  //     // const wrapper = mount(<span>test</span>)
  //     expect(wrapper.find('.c7n-pro-modal-content').length).toBe(1);
  //   })
  // });

  describe('queryDataSet', () => {
    it('queryDataSet-校验是否可以成功创建查询数据源', () => {
      const queryDs = new DataSet({
        autoCreate: true,
        fields: commonQueryFields,
      })
      const ds = new DataSet({ ...commonDs, queryDataSet: queryDs });
      expect(ds.queryDataSet).toBeInstanceOf(DataSet);
      expect(ds.queryDataSet.get(0)).toBeInstanceOf(Record);
      expect(ds.queryDataSet.get(0).get('name')).toBe('Hugh');
      expect(ds.queryDataSet.length).toBe(1);
    })
  });
  describe('fields', () => {
    it('fields-校验是否可以去设置字段属性数组', () => {
      const ds = new DataSet({ ...commonDs });
      commonDs.fields.map((eachFieldProp) => {
        expect(ds.fields.has(eachFieldProp.name)).toBeTruthy();
      })
    })
  });
  describe('queryFields', () => {
    it('设置queryFields-不设置queryDataSet-校验是否可以去设置查询字段属性数组', () => {
      const ds = new DataSet({
        ...commonDs,
        queryFields: commonQueryFields
      });
      expect(ds.queryDataSet).toBeInstanceOf(DataSet);
      commonQueryFields.map((eachFieldProp) => {
        expect(ds.queryDataSet.fields.has(eachFieldProp.name)).toBeTruthy();
      })
    })
    it('设置queryFields-设置queryDataSet-验证优先级', () => {
      const queryDataSetFields = [
        { name: 'enable1', type: 'boolean', label: '是否开启', defaultValue: true },
        { name: 'name1', type: 'string', label: '姓名', defaultValue: '通过queryDataSet设置' },
      ];
      const queryDs = new DataSet({
        autoCreate: true,
        fields: queryDataSetFields,
      })
      const ds = new DataSet({ ...commonDs, queryFields: commonQueryFields, queryDataSet: queryDs });
      
      expect(ds.queryDataSet).toBeInstanceOf(DataSet);
      commonQueryFields.map((eachFieldProp) => {
        expect(ds.queryDataSet.fields.has(eachFieldProp.name)).toBeTruthy();
      })
      // TODO: 查询字段的queryDateSet.length此处为0，但是官网为1
      // expect(ds.queryDataSet.get(0).get('name')).toBe(commonQueryFields[1].defaultValue);
    })
  });
})




