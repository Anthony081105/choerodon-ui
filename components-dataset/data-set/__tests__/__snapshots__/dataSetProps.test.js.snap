// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSetProps modifiedCheck modifiedCheck: true,有记录更改,验证是否有警告提示 1`] = `
<div
  class="c7n-pro-table-wrapper"
>
  <div
    class="c7n-spin-nested-loading"
  >
    <div
      class="c7n-spin-container"
    >
      <div
        class="c7n-pro-table c7n-pro-table-bordered"
        style="line-height: 0.3rem;"
        tabindex="0"
      >
        <div
          class="c7n-pro-table-content c7n-pro-table-content-overflow"
        >
          <table
            class="c7n-pro-table-last-row-bordered"
            style="width: 0.5rem;"
          >
            <colgroup>
              <col
                class="c7n-pro-table-col"
                style="min-width: 0.5rem; width: 0.5rem;"
              />
            </colgroup>
            <thead
              class="c7n-pro-table-thead c7n-pro-table-column-resizable"
            >
              <tr>
                <th
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-cell-last c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="text-align: center; left: 0px;"
                >
                  <div
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-row-height-fixed"
                    style="height: 0.3rem; line-height: 0.28rem;"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="c7n-pro-table-tbody"
            >
              <tr
                class="c7n-pro-table-row c7n-pro-table-row-current"
                data-index="1042"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1043"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1044"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1045"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1046"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1047"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1048"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1049"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1050"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
              <tr
                class="c7n-pro-table-row"
                data-index="1051"
                tabindex="-1"
              >
                <td
                  class="c7n-pro-table-cell c7n-pro-table-cell-fix-left c7n-pro-table-selection-column"
                  data-index="__selection-column__"
                  style="left: 0px;"
                >
                  <span
                    class="c7n-pro-table-cell-inner c7n-pro-table-cell-inner-bordered c7n-pro-table-cell-inner-row-height-fixed"
                    style="text-align: center; height: 0.3rem; line-height: 0.26rem;"
                    tabindex="-1"
                  >
                    <label
                      class="c7n-pro-checkbox-wrapper"
                    >
                      <input
                        class="c7n-pro-checkbox"
                        data-selection-key="__selection-column__"
                        type="checkbox"
                        value="true"
                      />
                      <i
                        class="c7n-pro-checkbox-inner"
                      />
                    </label>
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="c7n-pro-table-sticky-shadow"
          style="left: 0.49rem;"
        />
        <div
          class="c7n-pro-table-split-line"
        />
      </div>
    </div>
  </div>
  <nav
    class="c7n-pro-pagination-wrapper c7n-pro-table-pagination c7n-pro-pagination"
  >
    <span
      class="c7n-pro-pagination-perpage"
    >
      Number of items per page:
    </span>
    <span
      class="c7n-pro-select-wrapper c7n-pro-pagination-size-changer c7n-pro-select-sm c7n-pro-select-suffix-button c7n-pro-select-border c7n-pro-select-not-editable"
    >
      <input
        type="hidden"
        value="10"
      />
      <label>
        <input
          autocomplete="off"
          class="c7n-pro-select"
          readonly=""
          style="width: 4px; box-sizing: content-box;"
          type="text"
          value="10"
        />
        <div
          class="c7n-pro-select-suffix"
        >
          <i
            class="icon icon-baseline-arrow_drop_down c7n-pro-select-trigger"
          />
        </div>
      </label>
    </span>
    <span
      class="c7n-pro-pagination-page-info"
    >
      1 - 10 / 30
    </span>
    <button
      class="c7n-pro-btn-wrapper c7n-pro-pagination-pager c7n-pro-pagination-pager-first c7n-pro-btn-disabled c7n-pro-btn c7n-pro-btn-flat c7n-pro-btn-default c7n-pro-btn-icon-only"
      disabled=""
      type="button"
    >
      <i
        class="icon icon-first_page"
      />
    </button>
    <button
      class="c7n-pro-btn-wrapper c7n-pro-pagination-pager c7n-pro-pagination-pager-prev c7n-pro-btn-disabled c7n-pro-btn c7n-pro-btn-flat c7n-pro-btn-default c7n-pro-btn-icon-only"
      disabled=""
      type="button"
    >
      <i
        class="icon icon-navigate_before"
      />
    </button>
    <button
      class="c7n-pro-btn-wrapper c7n-pro-pagination-pager c7n-pro-pagination-pager-next c7n-pro-btn c7n-pro-btn-flat c7n-pro-btn-default c7n-pro-btn-icon-only"
      type="button"
    >
      <i
        class="icon icon-navigate_next"
      />
      <div
        class="c7n-ripple-wrapper"
      />
    </button>
    <button
      class="c7n-pro-btn-wrapper c7n-pro-pagination-pager c7n-pro-pagination-pager-last c7n-pro-btn c7n-pro-btn-flat c7n-pro-btn-default c7n-pro-btn-icon-only"
      type="button"
    >
      <i
        class="icon icon-last_page"
      />
      <div
        class="c7n-ripple-wrapper"
      />
    </button>
  </nav>
</div>
`;
